mech_client/interact.py

```diff
@@ -138,20 +180,28 @@ def watch_for_events(
                 no_receipt = False
             except Exception:
                 time.sleep(1)
-        if count == 0:
+        event_signature = tx_receipt["logs"][0]["topics"][0].hex()
+        if event_signature == EVENT_SIGNATURE_REQUEST:
             rich_logs = contract_instance.events.Request().processReceipt(tx_receipt)
             request_id = rich_logs[0]["args"]["requestId"]
             print(f"Request on-chain with id: {request_id}")
-        if count == 1:
+        if event_signature == EVENT_SIGNATURE_DELIVER:
             rich_logs = contract_instance.events.Deliver().processReceipt(tx_receipt)
             data = rich_logs[0]["args"]["data"]
+            request_id_ = rich_logs[0]["args"]["requestId"]
+            if request_id != request_id_:
```

Comment:
so the assumption here is that there will always be a req before a deliver. What would happen if there's a deliver, request_id will be references before declared no?

Reply:
good point - fixed now